win32/winapp.cpp

```diff
@@ -214,7 +214,7 @@ oeWin32Application::oeWin32Application(const char *name, unsigned flags, HInstan
     oeWin32Application::first_time = false;
   }
 
-  HICON dicon = ExtractIcon((HINSTANCE)hinst, "descent 3.exe", 0);
+  HICON dicon = ExtractIcon(NULL, "descent 3.exe", 0);
```

Comment:
how are you sure `hinst` is always null? And prefer `nullptr` to `NULL` in C++

Reply:
roger, wasn't sure if it was still just windows-dev practice to pass `NULL` for `HINSTANCE`-esque handles

> how are you sure hinst is always null?

i'm actually fairly sure that it _isn't_. however, the `ExtractIconA` method header marks that parameter as `_Reserved_`, indicating that it should always be set to null:
```
SHSTDAPI_(HICON) ExtractIconA(_Reserved_ HINSTANCE hInst, _In_ LPCSTR pszExeFileName, UINT nIconIndex);
```
```
// Reserved pointer parameters, must always be NULL.
#define _Reserved_                      _SAL2_Source_(_Reserved_, (), _Pre1_impl_(__null_impl))
```
The result of passing a real value here was a `C6388 'hinst' might not be '0': this does not adhere to the specification for the function 'ExtractIconA'` warning.

Reply:
Ok I got it, I'm not too familiar with Windows APIs :)

Reply:
same lol. thankfully MSDN is Documentation Heaven